test/functional/connection.test.js

```diff
@@ -274,3 +274,55 @@ describe('Connection', function () {
     }
   });
 });
+
+it(
+  'should be able to connect again after close',
+  withClient(function (client, done) {
+    expect(client.isConnected()).to.be.true;
+
+    const collection = client.db('testReconnect').collection('test');
+    collection.insertOne({ a: 1 }, (err, result) => {
+      expect(err).to.not.exist;
+      expect(result).to.exist;
+
+      client.close(err => {
+        expect(err).to.not.exist;
+        expect(client.isConnected()).to.be.false;
+
+        client.connect(err => {
+          expect(err).to.not.exist;
+          expect(client.isConnected()).to.be.true;
+
+          collection.insertOne({ b: 2 }, (err, result) => {
+            expect(err).to.not.exist;
+            expect(result).to.exist;
+            expect(client.topology.isDestroyed()).to.be.false;
+            done();
```

Comment:
Can this instead be a find on the previous insertion instead of another insert I guess just to show the round trip of connecting back to the same mongodb with a collection instance from the previous connected context.